logger/logger.go

```diff
@@ -0,0 +1,50 @@
+// Package log provides a log interface
+package logger
+
+import (
+	"fmt"
+	"sync"
+)
+
+// Logger is a generic logging interface
+type Logger interface {
+	Init(options ...Option) error
+	// String returns the name of logger
+	String() string
+	// SetLevel updates the logging level.
+	SetLevel(Level)
+	// Level returns the logging level
+	Level() Level
+	// Log inserts a log entry.  Arguments may be handled in the manner
+	// of fmt.Print, but the underlying logger may also decide to handle
+	// them differently.
+	Log(level Level, v ...interface{})
+	// Logf insets a log entry.  Arguments are handled in the manner of
+	// fmt.Printf.
+	Logf(level Level, format string, v ...interface{})
+	// Logfi prints fields using JSON or KV.
+	Logfi(level Level, msg string, fields ...Field)
```

Comment:
I would remove Logfi since we have Fields. The idea being that when you want to log with fields you do log.Fields("foo", "bar").Log("my log message")

Reply:
OK. thats better